### PR TITLE
asusctl: 4.5.8 -> 4.6.2

### PR DESCRIPTION
**Pull request by aacebedo**
###### Description of changes

Update to the 4.6.2
There are breaking changes on the config files. No more toml or json files only ron (rusty object notation) files.
The format is not supported in formats.nix so I changed the value of the parameters to txt. 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
This is a breaking NixOS change. Maybe we can write a generator for RON?

**Comment by K900** [comment]
We should probably incorporate the changes from https://github.com/NixOS/nixpkgs/pull/225326, and maybe patch out the RON conversion for now? It seems to still read JSON/TOML configuration, we just need it to stop writing RON files back out.

**Comment by K900** [comment]
I've also reached out to upstream and we'll see what we can come up with.

**Comment by K900** [comment]
Talked to upstream maintainers last night, they are intending to only support RON going forward, which makes our job difficult, as RON is hard to encode in Nix in an obvious way. I think we'll have to make do with raw strings to make branch-off.

**Comment by K900** [comment]
The freeze for breaking NixOS changes is today, so I guess we have to merge this as-is (which is to say, without a RON generator). Can you please add a release note documenting the config format change?

File: nixos/modules/services/hardware/asusd.nix [hunk 1]
```diff
@@ -19,55 +17,55 @@ in
       };
 
       animeConfig = lib.mkOption {
-        type = json.type;
-        default = { };
+        type = lib.types.nullOr lib.types.str;
+        default = null;
         description = lib.mdDoc ''
-          The content of /etc/asusd/anime.conf.
+          The content of /etc/asusd/anime.ront.
```
Comment (K900) [inline nixos/modules/services/hardware/asusd.nix:23]: Also typo here

Comment (aacebedo) [reply nixos/modules/services/hardware/asusd.nix]: fixed

Comment (K900) [reply nixos/modules/services/hardware/asusd.nix]: Thanks! Can you also add the release note? Somewhere around https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md#backward-incompatibilities-sec-release-2305-incompatibilities

**Comment by aacebedo** [comment]
> The freeze for breaking NixOS changes is today, so I guess we have to merge this as-is (which is to say, without a RON generator). Can you please add a release note documenting the config format change?

I tested it it seems to work on my side. I also integrated changes of #225326 too. Do you think we can merge it?

**Comment by K900** [comment]
The changes are OK, but we need to add a release note to inform people of the breakage.

**Comment by aacebedo** [comment]
> The changes are OK, but we need to add a release note to inform people of the breakage.

Where do we do that?

**Comment by K900** [comment]
https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md#backward-incompatibilities-sec-release-2305-incompatibilities

**Comment by aacebedo** [comment]
> https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md#backward-incompatibilities-sec-release-2305-incompatibilities

Done. Tell me what you think of the addition.

**Comment by K900** [comment]
Some nitpicks:
- it's usually better to refer to the options by full path, e.g. `services.asusd` and not just `asusd`, so users can search the option name in the changelog
- the common term for "JSON based" in NixOS is "structured configuration"
- not supporting RON is more of a Nixpkgs issue than a Nix issue

I'd write something like

"`services.asusd` configuration now uses strings instead of structured configuration, as upstream switched to the [RON](https://github.com/ron-rs/ron) configuration format. Support for structured configuration may return when RON generation is implemented in nixpkgs."

**Comment by aacebedo** [comment]
> Some nitpicks:
> 
> * it's usually better to refer to the options by full path, e.g. `services.asusd` and not just `asusd`, so users can search the option name in the changelog
> * the common term for "JSON based" in NixOS is "structured configuration"
> * not supporting RON is more of a Nixpkgs issue than a Nix issue
> 
> I'd write something like
> 
> "`services.asusd` configuration now uses strings instead of structured configuration, as upstream switched to the [RON](https://github.com/ron-rs/ron) configuration format. Support for structured configuration may return when RON generation is implemented in nixpkgs."

way better thanks. I updated the file.

**Comment by K900** [comment]
Thanks!

**Pull request merged**